metagraph/src/common/vectors/bitmap_builder.cpp

```diff
@@ -0,0 +1,37 @@
+#include "bitmap_builder.hpp"
+
+#include "common/elias_fano.hpp"
+
+
+const uint64_t kStreamBufferSize = 100'000;
+
+
+bitmap_builder_set_disk::InitializationData
+bitmap_builder_set_disk::get_initialization_data() const {
+    if (merged_)
+        throw std::runtime_error("ERROR: Trying to flush the bitmap twice");
+
+    mtg::common::EliasFanoEncoderBuffered<uint64_t> encoder(tmp_dir_/"merged",
+                                                            kStreamBufferSize);
+    auto &merged = const_cast<bitmap_builder_set_disk*>(this)->set_bit_positions_.data();
+    uint64_t num_set_bits = 0;
+    for (auto &it = merged.begin(); it != merged.end(); ++it) {
```

Comment:
This `const_cast` looks kind of ugly and is unnecessary since you're not using move iterators. Maybe add a `const_iterator` to this class to avoid it?

Reply:
ChunkedWaitQueue hasn't got a const iterator. That's why we need to cast.

Reply:
why not add one? it's only a few lines of boilerplate code

Reply:
Because it actually does change the state of the queue, hence it's not const.
One needs to move the merging logic into the iterator in order to fix this, and that's more work.